### ISSUE TITLE
Detect useless `map`

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#4305]
The following code could probably be flagged as using `map` when it should just use `each`.

```ruby
def prune_inactive_images
  images = Image.where("last_viewed_at > ?", 2.years.ago)

  images.map do |image|
    puts "Deleting image #{image.name}"
  end

  images.count
end
```

* The `map` result isn't assigned
* It isn't the last statement in the method

**Comment by bbatsov** [comment]
Seems like a good idea to me.

**Comment by Drenmi** [comment]
Seems very reasonable to me. 🙂 Indicating that you will use the return value and then not using it is confusing.

One thing to watch out for is that the statement following isn't a conditional.

**Comment by rrosenblum** [comment]
> One thing to watch out for is that the statement following isn't a conditional.

I think the bigger thing here is that the return of `map` is not being assigned to anything and the `map` statement isn't the last line of the method.

I think the same thing can be said for `find` and `select` instead of `any?` and `reject` instead of `none?`. These tend to be scoped to being used in if statements.

```ruby
if foo.find { |f| f == 1 }
  something
end
```

**Comment by koic** [comment]
When implementing this cop, I think that it should be considered that method chain is not used. It is a similar reason that the return value is not used.
